### DISCUSSION TITLE
CefSharp.WinForms plugin stops working after adding "enable-features" configuration

### DISCUSSION THREAD

**Discussion by k4mikazzeeHD** [cefsharp/CefSharp#4739] [General]
I am encountering an issue with my plugin developed using Class Library where I'm integrating CefSharp.WinForms to connect to the web. Initially, everything works as expected, and the browser window opens without any trouble.

![Captura de tela 2024-03-02 151546](https://github.com/cefsharp/CefSharp/assets/5115988/12f1495a-1bd1-4008-a082-4834be786140)

However, I need to adjust a specific configuration to connect to Google Earth (without this configuration, earth never loads):

`settings.CefCommandLineArgs.Add("enable-features", "SharedArrayBuffer");`

Unfortunately, after adding this line, the functionality of my plugin ceases to work entirely.

My code:

`public partial class Form1 : Form
{
    public CefSharp.WinForms.ChromiumWebBrowser chromeBrowser;
    public Form1()
    {
        string cefSharpWinFormsPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "CefSharp.WinForms.dll");
        Assembly.LoadFrom(cefSharpWinFormsPath);

        InitializeComponent();

        InitializeChromium();
    }

    private void Form1_Load(object sender, EventArgs e)
    {

    }

    private void InitializeChromium()
    {
        var settings = new CefSharp.WinForms.CefSettings();
        settings.CefCommandLineArgs.Add("enable-features", "SharedArrayBuffer");
        Cef.Initialize(settings);
        chromeBrowser = new CefSharp.WinForms.ChromiumWebBrowser("https://earth.google.com/");
        this.Controls.Add(chromeBrowser);
        chromeBrowser.Dock = DockStyle.Fill;
    }

    private void Form1_FormClosing(object sender, FormClosingEventArgs e)
    {
        Cef.Shutdown();
    }
}`

Am i doing something wrong?

As I mentioned, removing the settings line restores functionality to the plugin. However, any attempt to alter the configuration, regardless of the specific change made, causes the plugin to malfunction.

**Comment by amaitland** [comment]
Also asked at https://stackoverflow.com/questions/78093866/cefsharp-winforms-plugin-stops-working-after-adding-enable-features-configurat

Please just as your question once in future.

---

Check the log file for errors https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

As it's not a feature that's enabled by default it's not something that is tested in the context of CEF.

You'll likely need to search/ask on https://www.magpcss.org/ceforum/index.php
